src/neural/loader.cc

```diff
@@ -108,6 +108,20 @@ void FixOlderWeightsFile(WeightsFile* file) {
   using nf = pblczero::NetworkFormat;
   auto network_format = file->format().network_format().network();
   const auto has_network_format = file->format().has_network_format();
+  // Hack for old encoding compatibility. REMOVE BEFORE RELEASE.
+  if (has_network_format &&
+      network_format == pblczero::NetworkFormat::NETWORK_SE_WITH_HEADFORMAT &&
+      file->weights().encoder().size() > 0) {
+    CERR << "Attention body detected, hacking network format.";
+    auto* net = file->mutable_format()->mutable_network_format();
+    net->set_network(
+        pblczero::NetworkFormat::NETWORK_ATTENTIONBODY_WITH_HEADFORMAT);
+    if (file->weights().has_smolgen_w()) {
+      CERR << "BT2 detected, hacking activations.";
```

Comment:
Maybe change this to "Smolgen detected, ...", since we may likely see other nets (like masterkni's distilled t1 256x10 net), that still have old proto, but are not BT2.